src/dotnet-core-uninstall/Shared/VSVersioning/VisualStudioSafeVersionsExtractor.cs

```diff
@@ -113,13 +113,17 @@ public static Dictionary<Bundle, string> GetReasonRequiredStrings(IEnumerable<Bu
             foreach (var division in bundlesByDivisions)
             {
                 var requiredBundle = division.Key.Max();
-                requirementStringResults = requirementStringResults.Append((requiredBundle, division.Value));
-                requirementStringResults = requirementStringResults.Concat(division.Key
-                    .Where(bundle => !bundle.Equals(requiredBundle))
-                    .Select(bundle => (bundle, string.Empty)));
+                requirementStringResults = requirementStringResults.Concat([
+                    (requiredBundle, division.Value),
+                    ..division.Key
+                        .Where(bundle => !bundle.Equals(requiredBundle))
+                        .Select(bundle => (bundle, string.Empty))
+                ]);
             }
 
             return requirementStringResults
+                .GroupBy(pair => pair.bundle)
+                .Select(group => group.First()) // Remove duplicates
```

Comment:
Why does ordering matter in a dictionary? I think these are real duplicates and can't have different values. Is that correct?

Reply:
Yes, although potential duplicates might arise as well from unforseen cases with the regex. So just playing it safe just in case as this exception would not be handled